src/cargo/util/toml/mod.rs

```diff
@@ -1513,6 +1514,32 @@ impl TomlManifest {
     }
 }
 
+/// Returns the name of the README file for a `TomlProject`.
+fn readme_for_project(package_root: &Path, project: &TomlProject) -> Option<String> {
+    match &project.readme {
+        None => default_readme_from_package_root(package_root),
+        Some(value) => match value {
+            StringOrBool::Bool(false) => None,
+            StringOrBool::Bool(true) => default_readme_from_package_root(package_root),
```

Comment:
This is an interesting case because `readme = true` could actually still infer no actual README if the files are missing. Perhaps for now we could just make `readme = true` an error? Otherwise we probably want to assert that the default readme is indeed present, and `None` isn't returned in this case.

Reply:
Sorry for the delay in getting back to this.

What do you think of this approach: Similar to how the `build` field is handled, if `readme = true`, we assume a default value of `README.md` (instead of possibly `None`, as you pointed out). Then, we can validate that the file really exists while creating the `ManifestMetadata`, and `bail!` if the file doesn't exist. This would be more consistent with the behavior of the `build` field, but I'm not sure how important that is.

Reply:
Yeah that sounds reasonable to me!